fbw-a380x/src/systems/instruments/src/MFD/pages/common/InputField.tsx

```diff
@@ -0,0 +1,604 @@
+      this.props.tmpyActive.sub((v) => {
+        if (v === true) {
+          this.textInputRef.instance.classList.add('tmpy');
+        } else {
+          this.textInputRef.instance.classList.remove('tmpy');
+        }
+      }, true),
+    );
+
+    if (this.props.dataEntryFormat.reFormatTrigger) {
+      this.subs.push(this.props.dataEntryFormat.reFormatTrigger.sub(() => this.updateDisplayElement()));
+    }
+
+    this.textInputRef.instance.addEventListener('keypress', (ev) => this.onKeyPress(ev));
+    this.textInputRef.instance.addEventListener('keydown', (ev) => this.onKeyDown(ev));
+
+    if (!this.props.handleFocusBlurExternally) {
+      this.textInputRef.instance.addEventListener('focus', () => this.onFocus());
+      this.textInputRef.instance.addEventListener('blur', () => {
+        this.onBlur();
+      });
+      this.spanningDivRef.instance.addEventListener('click', () => {
+        this.textInputRef.instance.focus();
+      });
+      this.leadingUnitRef.instance.addEventListener('click', () => {
+        this.textInputRef.instance.focus();
+      });
+      this.trailingUnitRef.instance.addEventListener('click', () => {
+        this.textInputRef.instance.focus();
+      });
+    }
+
+    this.props.hEventConsumer.handle((key) => {
+      if (!this.isFocused.get()) {
+        return;
+      }
+
+      // Un-select the text
+      this.textInputRef.instance.classList.remove('valueSelected');
+
+      if (key.match(/^[a-zA-Z0-9]{1}$/)) {
+        this.handleKeyInput(key);
+      }
+
+      if (key === 'ENT') {
+        this.handleEnter();
+      }
+
+      if (key === 'SP') {
+        this.handleKeyInput(' ');
+      }
+
+      if (key === 'SLASH') {
+        this.handleKeyInput('/');
+      }
+
+      if (key === 'DOT') {
+        this.handleKeyInput('.');
+      }
+
+      if (key === 'PLUSMINUS') {
+        const val = this.modifiedFieldValue.get();
+        if (val && val.substring(0, 1) === '+') {
+          this.modifiedFieldValue.set(`-${val.substring(1)}`);
+        } else if (val && val.substring(0, 1) === '-') {
+          this.modifiedFieldValue.set(`+${val.substring(1)}`);
+        } else {
+          this.modifiedFieldValue.set(`-${this.modifiedFieldValue.get()}`);
+        }
+      }
+
+      if (key === 'BACKSPACE') {
+        this.handleBackspace();
+      }
+
+      if (key === 'ESC' || key === 'ESC2') {
+        const [formatted] = this.props.dataEntryFormat.format(this.props.value.get());
+        this.modifiedFieldValue.set(formatted);
+        this.handleEnter();
+      }
+
+      if (key === 'UP' || key === 'RIGHT' || key === 'DOWN' || key === 'LEFT') {
+        // Unsupported atm
+      }
+    });
+
+    /* if (this.props.inViewEvent) {
+            this.subs.push(this.props.inViewEvent.whenChanged().handle((inView) =>
+            {
+                console.warn('inView: ' + inView);
+                if (!inView) {
+                    this.onBlur();
+                }
+            }));
+        } */
```

Comment:
Remove this is it's no longer needed.

Reply:
This is preparation for automatic un-focusing if the node isn't in view anymore. We would need changes in the model for that though, won't happen in the near future. I added a comment.